src/systems/a320_systems/src/pneumatic.rs

```diff
@@ -1,27 +1,3039 @@
+        engines: [&(impl EngineCorrectedN1 + EngineCorrectedN2); 2],
+        overhead_panel: &mut A320PneumaticOverheadPanel,
+        engine_fire_push_buttons: &impl EngineFirePushButtons,
+        hydraulics: &A320Hydraulic,
+    ) {
+        self.apu.update(&self.apu_bleed_air_controller);
+
+        for bleed_monitoring_computer in self.bleed_monitoring_computers.iter_mut() {
+            bleed_monitoring_computer.update(
+                context,
+                &self.engine_systems,
+                self.apu_bleed_air_valve.is_open(),
+                overhead_panel,
+                engine_fire_push_buttons,
+                self.cross_bleed_valve.is_open(),
+                &self.fadec,
+            );
+
+            // I am not exactly sure if both BMCs should actually control this valve all the time.
+            self.cross_bleed_valve
+                .update_open_amount(&bleed_monitoring_computer.main_channel);
+        }
+
+        let (bleed_monitoring_computer_one, bleed_monitoring_computer_two) =
+            self.bleed_monitoring_computers.split_at_mut(1);
+
+        bleed_monitoring_computer_one[0].check_for_failure(&mut bleed_monitoring_computer_two[0]);
+        bleed_monitoring_computer_two[0].check_for_failure(&mut bleed_monitoring_computer_one[0]);
+
+        for engine_starter_valve_controller in self.engine_starter_valve_controllers.iter_mut() {
+            engine_starter_valve_controller.update(&self.fadec);
+        }
+
+        for engine_system in self.engine_systems.iter_mut() {
+            for bleed_monitoring_computer in self.bleed_monitoring_computers.iter() {
+                let index = engine_system.number - 1;
+
+                // If we get an actual channel here, this means that the channel is not in slave mode
+                if let Some(channel) =
+                    bleed_monitoring_computer.channel_for_engine(engine_system.number)
+                {
+                    engine_system.update(
+                        context,
+                        channel,
+                        channel,
+                        &self.engine_starter_valve_controllers[index],
+                        channel,
+                        engines[index],
+                    );
+                }
+            }
+        }
+
+        self.update_hydraulic_reservoir_spatial_volumes(
+            hydraulics.green_reservoir_capacity() - hydraulics.green_reservoir_volume(),
+            hydraulics.blue_reservoir_capacity() - hydraulics.blue_reservoir_volume(),
+            hydraulics.yellow_reservoir_capacity() - hydraulics.yellow_reservoir_volume(),
+        );
+
+        let (left_system, right_system) = self.engine_systems.split_at_mut(1);
+        self.apu_bleed_air_valve
+            .update_move_fluid(context, &mut self.apu, &mut left_system[0]);
+
+        self.cross_bleed_valve.update_move_fluid(
+            context,
+            &mut left_system[0],
+            &mut right_system[0],
+        );
```

Comment:
I have read this comment on the PR:
> While the bleed system has the logic for engine starting in place, this is not actually connected to the engines yet. This also means that crossbleed engine starts are not yet possible

**question**: I assume this is the place where I can see this isn't implemented yet. What are you thoughts on the implementation? Will it be the same lines of code, but just left and right reversed?

Reply:
Ah, the comment does not make this clear, but I was talking about the fact that the bleed system does pump air into the engines at the correct time, as can be seen here:

https://github.com/BlueberryKing/a32nx/blob/0ba61c788d1fe0d64548dc620b20f36257123337/src/systems/a320_systems/src/pneumatic.rs#L920-L926

But there is no logic in the Engine/C++ code to check this pressure apart from it directly checking whether the APU bleed is on.